File: nim/library-with-tests/tests/test_greet.nim
## Unit tests for the library's ``greet`` proc.

import std/unittest

import nim_library_with_tests_example

suite "greet":
  test "greets the named subject":
    check greet("world") == "hello, world"

  test "round-trips an empty string":
    check greet("") == "hello, "

File: rust/library-with-tests/tests/test_greet.rs
//! Integration test for the ``greet`` function.

use rust_library_with_tests_example::greet;

#[test]
fn integration_greets_world() {
    assert_eq!(greet("world"), "hello, world");
}

#[test]
fn integration_greets_empty() {
    assert_eq!(greet(""), "hello, ");
}
